pkg/core/blockchain.go

```diff
@@ -741,37 +862,208 @@ func (bc *Blockchain) removeOldTransfers(index uint32) time.Duration {
 	var removed, kept int64
 	var ts = h.Timestamp
 	prefixes := []byte{byte(storage.STNEP11Transfers), byte(storage.STNEP17Transfers)}
+	type kvTrimmed struct {
+		kv                      *storage.KeyValue
+		nextOlderLog            *storage.KeyValue
+		trimmed                 bool
+		newestTransferTimestamp uint64
+	}
+	var (
+		preEdge  func(batchAcc util.Uint160, k, v []byte, isEdge bool) bool
+		postEdge func(batchAcc util.Uint160, k, v []byte) bool
+		edgeLogs = make(map[util.Uint160]*kvTrimmed)
```

Comment:
Suppose you're working with the mainnet.

Reply:
Ugh, that's a tricky question, I think that the code still has this bottleneck (it's marked as TODO currently) during storage items handling, and we probably need some kind of intermediate DB flushes, but I'm not sure how to properly implement it.

Reply:
Fixed.